ev3dev2/motor.py

```diff
@@ -1774,6 +1791,48 @@ def _block(self):
         self.wait_until_not_moving()
 
 
+# line follower classes
+class LineFollowError(Exception):
+    """
+    Raised when a line following robot has lost the line
+    """
+    pass
+
+
+class LineFollowErrorTooFast(Exception):
+    """
+    Raised when a line following robot has been asked to follow
+    a line at an unrealistic speed
+    """
+    pass
+
+
+# line follower functions
+def follow_for_forever(_tank):
+    """
+    ``_tank``: the MoveTank object that is following a line
+    """
+    return True
+
+
+def follow_for_ms(_tank, ms):
+    """
+    ``_tank``: the MoveTank object that is following a line
+    ``ms`` : the number of milliseconds to follow the line
+    """
+    global stopwatch
```

Comment:
this is scary... can the tank object store a stopwatch which is started when line following is started and provide an accessor?

Reply:
We could pass a stopwatch object (one that is created at the start of `follow_line`) to the `follow_for_XYZ` function...I like that more than the global variable :)